### PR TITLE
EN6-141 advanced filtering for content

### PR DESCRIPTION
**Pull request by BrenoQVDS**
Creating filters for restriction and groups. I also added restriction to the DB

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
The tests will work only when the PR is accepted and merged: https://github.com/entando/entando-core/pull/834

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/AbstractContentSearcherDAO.java [hunk 1]
```diff
@@ -62,24 +63,29 @@ protected String getTableFieldName(String metadataFieldKey) {
...
+		}else if (metadataFieldKey.equals(IContentManager.CONTENT_GROUP_FILTER_KEY)) {
+			return "group";
 		} else throw new RuntimeException("Chiave di ricerca '" + metadataFieldKey + "' non riconosciuta");
 	}
 	
-	protected PreparedStatement buildStatement(EntitySearchFilter[] filters, 
+	protected PreparedStatement buildStatement(EntitySearchFilter[] filters,
 			Collection<String> userGroupCodes, boolean selectAll, Connection conn) {
-		return this.buildStatement(filters, null, false, userGroupCodes, selectAll, conn);
+		return this.buildStatement(filters, null,false, userGroupCodes, selectAll, conn);
```
Comment (ffleandro) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/AbstractContentSearcherDAO.java]: not necessary

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by BrenoQVDS** [comment]
Restest this please

**Comment by entando-service** [comment]
All is Well

**Pull request merged**